### ISSUE TITLE
Infinite loop when I check JSON schema file

### ISSUE THREAD

**Issue by mrix** [justinrainbow/json-schema#185]
I want to validate my JSON schema file
```php
$refResolver = new \JsonSchema\RefResolver();
$refResolver::$maxDepth = 10000;

$jsonSchema = $refResolver->getUriRetriever()->retrieve('http://json-schema.org/draft-04/schema#');
$refResolver->resolve($jsonSchema);
```

Script failed with error:
```
PHP Fatal error:  Uncaught exception 'JsonSchema\Exception\JsonDecodingException' with message 'The maximum stack depth has been exceeded' in /sync/vendor/justinrainbow/json-schema/src/JsonSchema/RefResolver.php:100
Stack trace:
#0 /sync/vendor/justinrainbow/json-schema/src/JsonSchema/RefResolver.php(63): JsonSchema\RefResolver->resolve(Object(stdClass))
#1 /sync/vendor/justinrainbow/json-schema/src/JsonSchema/RefResolver.php(208): JsonSchema\RefResolver->fetchRef('#', 'http://json-sch...')
#2 /sync/vendor/justinrainbow/json-schema/src/JsonSchema/RefResolver.php(114): JsonSchema\RefResolver->resolveRef(Object(stdClass), 'http://json-sch...')
#3 /sync/vendor/justinrainbow/json-schema/src/JsonSchema/RefResolver.php(152): JsonSchema\RefResolver->resolve(Object(stdClass), 'http://json-sch...')
#4 /sync/vendor/justinrainbow/json-schema/src/JsonSchema/RefResolver.php(126): JsonSchema\RefResolver->resolveArray in /sync/vendor/justinrainbow/json-schema/src/JsonSchema/RefResolver.php on line 100
```

**Comment by jojo1981** [comment]
Should also be fixed after merging: #245

**Comment by bighappyface** [comment]
Closed with https://github.com/justinrainbow/json-schema/pull/245
